From 003f26fd59e037a408f551dd7609e69642d00ffa Mon Sep 17 00:00:00 2001
From: Harald Sitter <sitter@kde.org>
Date: Thu, 17 Sep 2020 16:07:16 +0200
Subject: [PATCH] make c++11 standard compliant

static constexpr data members are not implicitly inline (they are starting with c++17)

https://github.com/AppImage/libappimage/issues/147
---
 src/libappimage/utils/StringSanitizer.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/libappimage/utils/StringSanitizer.h b/src/libappimage/utils/StringSanitizer.h
index 5301ec1..3059fcf 100644
--- a/src/libappimage/utils/StringSanitizer.h
+++ b/src/libappimage/utils/StringSanitizer.h
@@ -11,16 +11,16 @@ private:
     std::string input_;
 
     // these three lists can be used to compose alphabets for sanitization
-    static constexpr std::initializer_list<std::string::value_type> asciiLetters_ = {
+    inline static constexpr std::initializer_list<std::string::value_type> asciiLetters_ = {
         'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm',
         'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z',
         'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M',
         'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z',
     };
-    static constexpr std::initializer_list<std::string::value_type> asciiDigits_ = {
+    inline static constexpr std::initializer_list<std::string::value_type> asciiDigits_ = {
         '0', '1', '2', '3', '4', '5', '6', '7', '8', '9',
     };
-    static constexpr std::initializer_list<std::string::value_type> pathSafeChars_ = {
+    inline static constexpr std::initializer_list<std::string::value_type> pathSafeChars_ = {
         '.', '-', '_'
     };
 
-- 
2.25.1

